map/framework.cpp

```diff
@@ -3313,3 +3313,73 @@ void Framework::OnPowerSchemeChanged(power_management::Scheme const actualScheme
   if (actualScheme == power_management::Scheme::EconomyMaximum && GetTrafficManager().IsEnabled())
     GetTrafficManager().SetEnabled(false);
 }
+
+bool Framework::ShouldShowProducts() const
+{
+  auto const connectionStatus = Platform::ConnectionStatus();
+  if (connectionStatus == Platform::EConnectionType::CONNECTION_NONE)
+    return false;
+
+  std::string donateUrl;
+  if (!settings::Get(settings::kDonateUrl, donateUrl)) // donation is disabled
+    return false;
+
+  if (!storage::IsPointCoveredByDownloadedMaps(GetCurrentPlacePageInfo().GetMercator(), m_storage, * m_infoGetter))
+    return false;
+
+  static uint32_t kFirstLaunchTimeout;
+  static uint32_t kLastProductCloseTimeout;
+  static uint32_t kLastProductSelectTimeout;
+  #ifdef DEBUG
+  kFirstLaunchTimeout = 60 * 3; // 3 min because it may takes some time to download the map after the first launch in debug
+  kLastProductCloseTimeout = 10;
+  kLastProductSelectTimeout = 20;
+  #else
+  kFirstLaunchTimeout = 60 * 60 * 3; // 3 hours
+  kLastProductCloseTimeout = 60 * 60 * 24  * 14; // 14 days
+  kLastProductSelectTimeout = 180 * 24 * 60 * 60; // 180 days
+  #endif
```

Comment:
Better to declare constants in both cases, e.g:
```
#ifdef DEBUG
  uint32_t constexpr kFirstLaunchTimeout = 60 * 3; // 3 min because it may takes some time to download the map   #else
  uint32_t constexpr kFirstLaunchTimeout = 60 * 60 * 3; // 3 hours
```

Reply:
Thx. Fixed!